### PR TITLE
Add API HasRawData() for get vector

### PR DESCRIPTION
**Pull request by cydrain**

### CODE REVIEW

File: src/index/diskann/diskann.cc [hunk 1]
```diff
@@ -62,6 +62,15 @@ class DiskANNIndexNode : public IndexNode {
     expected<DataSetPtr, Status>
     GetVectorByIds(const DataSet& dataset, const Config& cfg) const override;
 
+    bool
+    HasRawData(const std::string& metric_type) const override {
+        if (metric_type == metric::L2) {
```
Comment (czs007) [inline src/index/diskann/diskann.cc:67]: just return metric_type == metric::L2?

Comment (cydrain) [reply src/index/diskann/diskann.cc:67]: ok

File: src/index/ivf/ivf.cc [hunk 2]
```diff
@@ -60,6 +60,21 @@ class IvfIndexNode : public IndexNode {
     RangeSearch(const DataSet& dataset, const Config& cfg, const BitsetView& bitset) const override;
     expected<DataSetPtr, Status>
     GetVectorByIds(const DataSet& dataset, const Config& cfg) const override;
+    bool
+    HasRawData(const std::string& metric_type) const override {
+        if constexpr (std::is_same<faiss::IndexIVFFlat, T>::value) {
```
Comment (czs007) [inline src/index/ivf/ivf.cc:65]: If none of the above conditions are met, will there be a return value?

File: src/index/ivf_raft/ivf_raft.cuh [hunk 3]
```diff
@@ -438,6 +438,16 @@ class RaftIvfIndexNode : public IndexNode {
...
 
+    virtual bool
+    HasRawData(const std::string& metric_type) const override {
+        if constexpr (std::is_same_v<detail::raft_ivf_flat_index, T>) {
+            return true;
+        }
+        if constexpr (std::is_same_v<detail::raft_ivf_pq_index, T>) {
+            return false;
+        }
+    }
```
Comment (czs007) [inline src/index/ivf_raft/ivf_raft.cuh:449]: If none of the above conditions are met, will there be a return value?

File: src/index/ivf/ivf.cc [hunk 4]
```diff
@@ -60,6 +60,21 @@ class IvfIndexNode : public IndexNode {
...
+        if constexpr (std::is_same<faiss::IndexIVFPQ, T>::value) {
+            return false;
+        }
+        if constexpr (std::is_same<faiss::IndexIVFScalarQuantizer, T>::value) {
+            return false;
+        }
+        if constexpr (std::is_same<faiss::IndexBinaryIVF, T>::value) {
+            return true;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (cydrain) [reply src/index/ivf/ivf.cc:77]: that will not happen

File: src/index/ivf_raft/ivf_raft.cuh (hunk 3 above)
Comment (cydrain) [reply src/index/ivf_raft/ivf_raft.cuh:449]: that will not happen

**Pull request merged**